service/controller/app/v1/resource/chartoperator/delete.go

```diff
@@ -0,0 +1,7 @@
+package chartoperator
+
+import "context"
+
+func (r Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
```

Comment:
Should we delete chart-operator if no `apps` are there?

Reply:
The resource in `cluster-operator` does not implement deletion but when there is an app and it is deleted whatever is managed with that app should be deleted IMO. Just to ensure app creation and deletion works.

Reply:
I think `EnsureDeleted` should be a no-op like it is here.

For me, it's a question of which operator is responsible. 

- Every TC needs a chart-operator which is what we are bootstrapping here.
- When a TC gets deleted aws-operator etc are responsible for deleting the TC and chart-operator in the TC goes away.

Reply:
One step back. Why do we have a separate resource for installing `chart-operator` at all? Why isn't it a generic operation to install whatever chart should be installed upon an `App` CR? Why is this special?

Reply:
`app-operator` watches `app CR and create `chart` CR related to that, but first of all, we need an operator that creating `chart` *CRD*. `chartoperator` would bring this chart-operator into the new TC.

Reply:
With you. It is for the bootstrapping.